include/zephyr/drivers/interrupt_controller/intc_vim.h

```diff
@@ -0,0 +1,383 @@
+#define VIM_GRP_FIQSTS_MSK_SHIFT             (0x00000000U)
+#define VIM_GRP_FIQSTS_MSK_RESETVAL          (0x00000000U)
+#define VIM_GRP_FIQSTS_MSK_MAX               (BIT_MASK(32))
+
+#define VIM_GRP_FIQSTS_RESETVAL              (0x00000000U)
+
+/* INTMAP */
+
+#define VIM_GRP_INTMAP_MSK_MASK              (BIT_MASK(32))
+#define VIM_GRP_INTMAP_MSK_SHIFT             (0x00000000U)
+#define VIM_GRP_INTMAP_MSK_RESETVAL          (0x00000000U)
+#define VIM_GRP_INTMAP_MSK_MAX               (BIT_MASK(32))
+
+#define VIM_GRP_INTMAP_RESETVAL              (0x00000000U)
+
+/* INTTYPE */
+
+#define VIM_GRP_INTTYPE_MSK_MASK             (BIT_MASK(32))
+#define VIM_GRP_INTTYPE_MSK_SHIFT            (0x00000000U)
+#define VIM_GRP_INTTYPE_MSK_RESETVAL         (0x00000000U)
+#define VIM_GRP_INTTYPE_MSK_MAX              (BIT_MASK(32))
+
+#define VIM_GRP_INTTYPE_RESETVAL             (0x00000000U)
+
+/* INT */
+
+#define VIM_PRI_INT_VAL_MASK                 (BIT_MASK(4))
+#define VIM_PRI_INT_VAL_SHIFT                (0x00000000U)
+#define VIM_PRI_INT_VAL_RESETVAL             (BIT_MASK(4))
+#define VIM_PRI_INT_VAL_MAX                  (BIT_MASK(4))
+
+#define VIM_PRI_INT_RESETVAL                 (BIT_MASK(4))
+
+/* INT */
+
+#define VIM_VEC_INT_VAL_MASK                 (0xFFFFFFFCU)
+#define VIM_VEC_INT_VAL_SHIFT                (0x00000002U)
+#define VIM_VEC_INT_VAL_RESETVAL             (0x00000000U)
+#define VIM_VEC_INT_VAL_MAX                  (BIT_MASK(30))
+
+#define VIM_VEC_INT_RESETVAL                 (0x00000000U)
+
+/* PID */
+
+#define VIM_PID_MINOR_MASK                   (BIT_MASK(6))
+#define VIM_PID_MINOR_SHIFT                  (0x00000000U)
+#define VIM_PID_MINOR_RESETVAL               (0x00000001U)
+#define VIM_PID_MINOR_MAX                    (BIT_MASK(6))
+
+#define VIM_PID_CUSTOM_MASK                  (0x000000C0U)
+#define VIM_PID_CUSTOM_SHIFT                 (0x00000006U)
+#define VIM_PID_CUSTOM_RESETVAL              (0x00000000U)
+#define VIM_PID_CUSTOM_MAX                   (BIT_MASK(2))
+
+#define VIM_PID_MAJOR_MASK                   (0x00000700U)
+#define VIM_PID_MAJOR_SHIFT                  (0x00000008U)
+#define VIM_PID_MAJOR_RESETVAL               (0x00000000U)
+#define VIM_PID_MAJOR_MAX                    (BIT_MASK(3))
+
+#define VIM_PID_RTL_MASK                     (0x0000F800U)
+#define VIM_PID_RTL_SHIFT                    (0x0000000BU)
+#define VIM_PID_RTL_RESETVAL                 (0x00000000U)
+#define VIM_PID_RTL_MAX                      (BIT_MASK(5))
+
+#define VIM_PID_FUNC_MASK                    (0x0FFF0000U)
+#define VIM_PID_FUNC_SHIFT                   (0x00000010U)
+#define VIM_PID_FUNC_RESETVAL                (0x00000090U)
+#define VIM_PID_FUNC_MAX                     (BIT_MASK(12))
+
+#define VIM_PID_BU_MASK                      (0x30000000U)
+#define VIM_PID_BU_SHIFT                     (0x0000001CU)
+#define VIM_PID_BU_RESETVAL                  (0x00000002U)
+#define VIM_PID_BU_MAX                       (BIT_MASK(2))
+
```

Comment:
These registers are not needed for your implementation, can you please clean this file up?
this was discussed in original PR as well : https://github.com/zephyrproject-rtos/zephyr/pull/59191#discussion_r1268906244